more_itertools/more.py

```diff
@@ -2121,8 +2137,76 @@ def rlocate(iterable, pred=bool):
     See :func:`locate` to for other example applications.
 
     """
-    try:
-        len_iter = len(iterable)
-        return (len_iter - i - 1 for i in locate(reversed(iterable), pred))
-    except TypeError:
-        return reversed(list(locate(iterable, pred)))
+    if window_size is None:
+        try:
+            len_iter = len(iterable)
+            return (
+                len_iter - i - 1 for i in locate(reversed(iterable), pred)
+            )
+        except TypeError:
+            pass
+
+    return reversed(list(locate(iterable, pred, window_size)))
+
+
+def replace(iterable, pred, substitutes, count=None, window_size=1):
+    """Yield the items from *iterable*, replacing the items for which *pred*
+    returns ``True`` with the items from the iterable *substitutes*.
+
+        >>> iterable = [1, 1, 0, 1, 1, 0, 1, 1]
+        >>> pred = lambda x: x == 0
+        >>> substitutes = (2, 3)
+        >>> list(replace(iterable, pred, substitutes))
+        [1, 1, 2, 3, 1, 1, 2, 3, 1, 1]
+
+    If *count* is given, the number of replacements will be limited:
+
+        >>> iterable = [1, 1, 0, 1, 1, 0, 1, 1]
+        >>> pred = lambda x: x == 0
+        >>> substitutes = [None]
+        >>> list(replace(iterable, pred, substitutes, count=2))
+        [1, 1, None, 1, 1, None, 1, 1]
```

Comment:
In this example, although count is "limited" to 2, **all** of the zeros are reolaced.  And doesn't capture the intent of `count` to limit replacements.  Consider `count=1` -> `[1, 1, None, 1, 1, 0, 1, 1]`

Reply:
Good point.